internal/dataplane/parser/parser.go

```diff
@@ -429,22 +431,17 @@ func getGatewayCerts(log logrus.FieldLogger, s store.Storer) []certWrapper {
 					if len(listener.TLS.CertificateRefs) > 1 {
 						// TODO support cert_alt and key_alt if there are 2 SecretObjectReferences
 						// https://github.com/Kong/kubernetes-ingress-controller/issues/2604
-						log.WithFields(logrus.Fields{
-							"gateway":  gateway.Name,
-							"listener": listener.Name,
-						}).Error("Gateway Listeners with more than one certificateRef are not supported")
+						p.registerTranslationFailure("listener '%s' has more than one certificateRef, it's not supported", gateway)
 						continue
 					}
 
 					ref := listener.TLS.CertificateRefs[0]
 
 					// SecretObjectReference is a misnomer; it can reference any Group+Kind, but defaults to Secret
 					if ref.Kind != nil {
-						if string(*ref.Kind) != "Secret" {
-							log.WithFields(logrus.Fields{
-								"gateway":  gateway.Name,
-								"listener": listener.Name,
-							}).Error("CertificateRefs to kinds other than Secret are not supported")
+						if kind := *ref.Kind; kind != "Secret" {
+							p.registerTranslationFailure(fmt.Sprintf("CertificateRefs to kinds other than Secret (%s) are not supported", kind), gateway)
```

Comment:
This is handled on the controller side (https://github.com/Kong/kubernetes-ingress-controller/blob/ec0f2702b99f8a75581e5c789bdf57a062ad1491/internal/controllers/gateway/gateway_utils.go#L462). Not possible to trigger this in the Parser.

Reply:
To confirm then, this and the other one that's handled controller-side should just be removed? Not sure why we'd keep the parser-side code and convert it to the registration system if it's effectively unreachable.

Reply:
Yeah, it seems to me we can drop both validations on the parser side. The logic verifying the reference permissions in the parser (`!newRefChecker(ref).IsRefAllowedByGrant(allowed)`) is duplicated in the `getReferenceGrantConditionReason` on the controller side: [e9bbb3a](https://github.com/Kong/kubernetes-ingress-controller/pull/3147/commits/e9bbb3a2a3850bda7ee17ee0d69abd58193bdd0c)

I've "moved" two unit tests that I found to be missing on the controller side from the parser suite (
8dc2ac6). Besides that, the ReferenceGrants compatibility is verified in the gateway conformance tests.